adyenv6b2ccheckoutaddon/acceleratoraddon/web/src/com/adyen/v6/controllers/pages/AdyenSummaryCheckoutStepController.java

```diff
@@ -178,6 +183,21 @@ public String placeOrder(@ModelAttribute("placeOrderForm") final PlaceOrderForm
             } catch (Exception e) {
                 LOGGER.error(ExceptionUtils.getStackTrace(e));
             }
+        } else if(PAYMENT_METHOD_POS.equals(adyenPaymentMethod)) {
+            try {
+                OrderData orderData = adyenCheckoutFacade.initiatePosPayment(cartData);
+                LOGGER.debug("Redirecting to confirmation!");
+                return redirectToOrderConfirmationPage(orderData);
+            } catch (ApiException e) {
+                LOGGER.error("API exception " + e.getError(), e);
+            } catch (AdyenNonAuthorizedPaymentException e) {
+                LOGGER.debug("AdyenNonAuthorizedPaymentException", e);
+                TerminalAPIResponse terminalApiResponse = e.getTerminalApiResponse();
+                ErrorConditionType refusalReason = getPosRefusalReason(terminalApiResponse);
```

Comment:
Refusal reasons will be used for response code REFUSED. We need to add response code checks for this. For example something like :
if (REFUSED == terminalApiResponse.getResultCode()) {
                    errorMessage = getErrorMessageByRefusalReason(terminalApiResponse.getRefusalReason());
                }

Reply:
Refactored.